services/ui-src/src/routes/Dashboard/Dashboard.tsx

```diff
@@ -52,6 +52,9 @@ export const Dashboard = () => {
   const [selectedReportId, setSelectedReportId] = useState<string | undefined>(
     undefined
   );
+  const [selectedReportFormData, setSelectedReportFormData] = useState<
+    AnyObject | undefined
+  >(undefined);
```

Comment:
Rather than setting and passing it to the modal separately, what if we changed the `setReportId` to not just take the ID but the entire report metadata object, so this would become something like below, then we'd just pass the whole thing in to the modal? 
```suggestion
  const [selectedReportMetadata, setSelectedReportMetadata] = useState<
    AnyObject | undefined
  >(undefined);
```